tests/test_request_header_authentication.py

```diff
@@ -0,0 +1,93 @@
+from unittest import mock  # noqa
+from werkzeug.test import EnvironBuilder
+
+from notifications_utils.request_helper import CustomRequest, _check_proxy_header_secret
+
+
+def test_request_header_contains_correct_secret():
+    builder = EnvironBuilder()
```

Comment:
These 3 lines could make a nice py.test fixture since they're repeated in most tests in this file. Alternatively, the tests are probably similar enough to be grouped and run with `@pytest.mark.parametrize`.

Reply:
Updated to use the `@pytest.mark.parametrize`. 

Although I'm not sure how I feel about adding logic to tests 🤔